torch/utils/data/sampler.py

```diff
@@ -212,7 +211,7 @@ class WeightedRandomSampler(Sampler[int]):
         [0, 1, 4, 3, 2]
     """
 
-    weights: Tensor
+    weights: 'torch.Tensor'
```

Comment:
Why? Just making it harder for the type checker.

Reply:
This is used to avoid cyclic imports. `torch.Tensor` is defined in `torch._tensor`.

```python
torch._tensor -> torch.utils.hooks -> torch.utils -> torch.utils.data
```

Reply:
@XuehaiPan Just import in an `if TYPE_CHECKING` block then

Reply:
It seems to be resolved by removing `torch.utils.hooks` in `torch._tensor`.